### PR TITLE
K8s: Ensure there is always a user in appcontext.User

### PR DESCRIPTION
**Pull request by ryantxu**
This is an alternative to: https://github.com/grafana/grafana/pull/78228

### CODE REVIEW

File: pkg/services/grafana-apiserver/auth/authorizer/provider.go [hunk 1]
```diff
@@ -17,12 +15,7 @@ func ProvideAuthorizer(
 	stackIDAuthorizer *stack.StackIDAuthorizer,
 	cfg *setting.Cfg,
 ) authorizer.Authorizer {
-	authorizers := []authorizer.Authorizer{
-		// This will allow privileged uses to do anything.
-		// In development mode, a privileged user is configured and saved into:
-		// ${data}/grafana-apiserver/grafana.kubeconfig
-		authorizerfactory.NewPrivilegedGroups(user.SystemPrivilegedGroup),
```
Comment (ryantxu) [inline pkg/services/grafana-apiserver/auth/authorizer/provider.go:24]: 🎉  -- this means we won't have the weird issues that system users can write to *any* namespaces even if totally invalid

**Review by DanCech — APPROVED** [review]
This seems like it should work

**Review by toddtreece — APPROVED** [review]

**Review by DanCech — APPROVED** [review]
Interesting approach!

**Comment by ryantxu** [comment]
The previous approach did not really work 😢 

The problem is that we can either run before or after the standard chain: https://github.com/kubernetes/kubernetes/blob/v1.28.3/staging/src/k8s.io/apiserver/pkg/server/config.go#L910

when running before, we do not have the k8s authn info, when running after -- the authorizers (that depend on appcontext.User) do not exist!

**Review by toddtreece — APPROVED** [review]
confirmed working with a stack ID set in custom.ini:
```
k apply -f pkg/tests/apis/playlist/testdata/playlist-test-create.yaml     
Error from server (Forbidden): error when retrieving current configuration of:
Resource: "playlist.grafana.app/v0alpha1, Resource=playlists", GroupVersionKind: "playlist.grafana.app/v0alpha1, Kind=Playlist"
Name: "test", Namespace: "default"
from server for: "pkg/tests/apis/playlist/testdata/playlist-test-create.yaml": playlists.playlist.grafana.app "test" is forbidden: User "system:apiserver" cannot get resource "playlists" in API group "playlist.grafana.app" in the namespace "default": wrong stack id is selected
```

**Pull request merged**